src/redux/data.ts

```diff
@@ -916,66 +924,82 @@ export const transactionsReceived = (
   message: TransactionsReceivedMessage | undefined,
   appended = false
 ) => async (
-  dispatch: ThunkDispatch<AppState, unknown, DataLoadTransactionSuccessAction>,
+  dispatch: ThunkDispatch<
+    AppState,
+    unknown,
+    DataLoadTransactionSuccessAction | DataUpdatePendingTransactionSuccessAction
+  >,
   getState: AppGetState
-) =>
-  withRunExclusive(async () => {
-    const isValidMeta = dispatch(checkMeta(message));
-    if (!isValidMeta) return;
-    const transactionData = message?.payload?.transactions ?? [];
-    if (appended) {
-      dispatch(checkForConfirmedSavingsActions(transactionData));
-    }
-    await dispatch(checkForUpdatedNonce(transactionData));
+) => {
+  const isValidMeta = dispatch(checkMeta(message));
+  if (!isValidMeta) return;
+  const transactionData = message?.payload?.transactions ?? [];
+  if (appended) {
+    dispatch(checkForConfirmedSavingsActions(transactionData));
+  }
+  await dispatch(checkForUpdatedNonce(transactionData));
 
-    const { accountAddress, nativeCurrency } = getState().settings;
-    const { purchaseTransactions } = getState().addCash;
-    const { transactions } = getState().data;
-    const { selected } = getState().wallets;
+  const { accountAddress, nativeCurrency } = getState().settings;
+  const { purchaseTransactions } = getState().addCash;
+  const { pendingTransactions, transactions } = getState().data;
+  const { selected } = getState().wallets;
 
-    let { network } = getState().settings;
-    if (network === Network.mainnet && message?.meta?.chain_id) {
-      network = message?.meta?.chain_id;
-    }
-    const {
-      parsedTransactions,
-      potentialNftTransaction,
-    } = await parseTransactions(
-      transactionData,
-      accountAddress,
-      nativeCurrency,
-      transactions,
-      purchaseTransactions,
-      network,
-      appended
-    );
-    if (appended && potentialNftTransaction) {
+  let { network } = getState().settings;
+  if (network === Network.mainnet && message?.meta?.chain_id) {
+    network = message?.meta?.chain_id;
+  }
+  const {
+    parsedTransactions,
+    potentialNftTransaction,
+  } = await parseTransactions(
+    transactionData,
+    accountAddress,
+    nativeCurrency,
+    transactions,
+    purchaseTransactions,
+    network,
+    appended
+  );
+  if (appended && potentialNftTransaction) {
+    setTimeout(() => {
+      dispatch(uniqueTokensRefreshState());
+    }, 60000);
+  }
+  const txHashes = parsedTransactions.map(({ hash }) => hash);
```

Comment:
we have a tool: `ethereumUtils.getHash(txn)` as our parsing from zerion txns sometimes includes `${txnHash}-0` as the hash